src/adaptation/ecap/MessageRep.cc

```diff
@@ -204,8 +204,8 @@ Adaptation::Ecap::RequestLineRep::uri(const Area &aUri)
 {
     // TODO: if method is not set, AnyP::Uri::parse will assume it is not connect;
     // Can we change AnyP::Uri::parse API to remove the method parameter?
-    const char *buf = aUri.toString().c_str();
-    const bool ok = theMessage.url.parse(theMessage.method, buf);
+    std::string uriBuffer = aUri.toString();
```

Comment:
Please use auto instead of std::string to avoid implicit type casting and unnecessary copies (two re-allocate cycles is bad enough).

Reply:
fixed

Reply:
> Please use auto instead of std::string to avoid implicit type casting and unnecessary copies (two re-allocate cycles is bad enough).

FYI: Yes, we should use `auto` when we can, including here, but there was no implicit cast here, and `auto` does not eliminate any copies in this context AFAICT: Area::toString() returns std::string which was the type used in the original PR code.